Projects/Server/Gumps/New/Components/GumpBuilder.cs

```diff
@@ -0,0 +1,698 @@
+            WriteEnd();
+        }
+
+        public void AddHtmlCentered(int x, int y, int width, int height, string? text, bool background = false, bool scrollbar = false)
+        {
+            WriteStart(Labels.Html);
+            WriteValue(x);
+            WriteValue(y);
+            WriteValue(width);
+            WriteValue(height);
+            WriteValueInternalized($"<CENTER>{text}</CENTER>");
+            WriteValue(background);
+            WriteValue(scrollbar);
+            WriteEnd();
+        }
+
+        public void AddHtmlCentered(int x, int y, int width, int height, ref GumpInterpolatedStringHandler<T, Centered> handler,
+            bool background = false, bool scrollbar = false)
+        {
+            WriteStart(Labels.Html);
+            WriteValue(x);
+            WriteValue(y);
+            WriteValue(width);
+            WriteValue(height);
+            WriteValueInternalized(ref handler);
+            WriteValue(background);
+            WriteValue(scrollbar);
+            WriteEnd();
+        }
+
+        public void AddHtmlCentered(int x, int y, int width, int height, int color, string? text, bool background = false, bool scrollbar = false)
+        {
+            WriteStart(Labels.Html);
+            WriteValue(x);
+            WriteValue(y);
+            WriteValue(width);
+            WriteValue(height);
+            WriteValueInternalized($"</BASEFONT COLOR=#{color:X6}><CENTER>{text}<CENTER></BASECOLOR>");
+            WriteValue(background);
+            WriteValue(scrollbar);
+            WriteEnd();
+        }
+
+        [SuppressMessage("Style", "IDE0060:Remove unused parameter", Justification = "used by interpolated string handler")]
+        public void AddHtmlCentered(int x, int y, int width, int height, int color,
+            [InterpolatedStringHandlerArgument(nameof(color))] ref GumpInterpolatedStringHandler<T, Centered> handler,
+            bool background = false, bool scrollbar = false)
+        {
+            WriteStart(Labels.Html);
+            WriteValue(x);
+            WriteValue(y);
+            WriteValue(width);
+            WriteValue(height);
+            WriteValueInternalized(ref handler);
+            WriteValue(background);
+            WriteValue(scrollbar);
+            WriteEnd();
+        }
+
+        public void AddHtmlLocalized(int x, int y, int width, int height, int number, bool background = false, bool scrollbar = false)
+        {
+            WriteStart(Labels.HtmlLocalized);
+            WriteValue(x);
+            WriteValue(y);
+            WriteValue(width);
+            WriteValue(height);
+            WriteValue(number);
+            WriteValue(background);
+            WriteValue(scrollbar);
+            WriteEnd();
+        }
+
+        public void AddHtmlLocalized(int x, int y, int width, int height, int number, short color, bool background = false, bool scrollbar = false)
+        {
+            WriteStart(Labels.HtmlLocalizedWithColor);
```

Comment:
Were you able to find out if doing this:
```suggestion
            WriteStart("xmfhtmlgumpcolor"u8);
```

Creates an allocation, or does it use a generated global constant?

Reply:
u8 strings are converted into global static byte array under the hood.
From my tests, the instruction:
```cs
AppendFake("test"u8);
```
Will be converted into these IL:
```
IL_0000: ldsflda valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=5' '<PrivateImplementationDetails>'::CEB73749C899693706EDE1E30C9929B3FD5DD926163831C2FB8BD41E6EFB1126
IL_0005: ldc.i4.4
IL_0006: newobj instance void valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>::.ctor(void*, int32)
// AppendFake(new byte[4] { 116, 101, 115, 116 });
IL_000b: call void TestClass::AppendFake(valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>)
```

So, the array is staticized into a global byte array variable with name "CEB73749C899693706EDE1E30C9929B3FD5DD926163831C2FB8BD41E6EFB1126" inside the hidden class <PrivateImplementationDetails>.
Everytime we use the same u8 string it will reuse the same static global variable and construct only a new ReadOnlySpan that reference it, like these statements:
```cs
AppendFake("test"u8);
AppendFake("test"u8);
```

That emits these:
```
IL_0000: ldsflda valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=5' '<PrivateImplementationDetails>'::CEB73749C899693706EDE1E30C9929B3FD5DD926163831C2FB8BD41E6EFB1126
IL_0005: ldc.i4.4
IL_0006: newobj instance void valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>::.ctor(void*, int32)
// AppendFake(new byte[4] { 116, 101, 115, 116 });
IL_000b: call void TestClass::AppendFake(valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>)
// AppendFake(new byte[4] { 116, 101, 115, 116 });
IL_0010: ldsflda valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=5' '<PrivateImplementationDetails>'::CEB73749C899693706EDE1E30C9929B3FD5DD926163831C2FB8BD41E6EFB1126
IL_0015: ldc.i4.4
IL_0016: newobj instance void valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>::.ctor(void*, int32)
IL_001b: call void TestClass::AppendFake(valuetype [System.Runtime]System.ReadOnlySpan`1<uint8>)
```